kart/sqlalchemy/gpkg.py

```diff
@@ -27,6 +25,42 @@ def _on_connect(pysqlite_conn, connection_record):
             dbcur.execute("PRAGMA foreign_keys = ON;")
             dbcur.execute(f"PRAGMA cache_size = -{cls.GPKG_CACHE_SIZE_MiB * 1024};")
 
+        path = os.path.expanduser(path)
         engine = sqlalchemy.create_engine(f"sqlite:///{path}", module=sqlite)
         sqlalchemy.event.listen(engine, "connect", _on_connect)
         return engine
+
+    @classmethod
+    def create_preparer(cls, engine):
+        return SQLiteIdentifierPreparer(engine.dialect)
+
+    @classmethod
+    def list_tables(cls, sess, db_schema=None, with_titles=False):
+        if db_schema is not None:
+            raise RuntimeError("GPKG files don't have a db_schema")
+
+        if with_titles:
+            gpkg_contents_exists = sess.scalar(
+                "SELECT COUNT(*) FROM sqlite_master WHERE type='table' AND name='gpkg_contents';",
+            )
+            if gpkg_contents_exists:
+                r = sess.execute(
+                    """
+                    SELECT SM.name, GC.identifier FROM sqlite_master SM
+                    LEFT OUTER JOIN gpkg_contents GC ON GC.table_name = SM.name
+                    WHERE SM.type='table'
+                    AND SM.name NOT LIKE 'sqlite%' AND SM.name NOT LIKE 'gpkg%' and SM.name NOT LIKE 'rtree%';
+                    """
```

Comment:
```suggestion
                    ORDER BY SM.name
                    """
```

these flow through to the table chooser prompt, so makes sense to get them in a pretty order

Reply:
Done